lib/models/device.coffee

```diff
@@ -291,16 +227,9 @@ exports.has = (name, callback) ->
 #		console.log("Is device online? #{isOnline}")
 ###
 exports.isOnline = (name, callback) ->
-
-	if not callback?
-		throw new errors.ResinMissingParameter('callback')
-
-	if not _.isFunction(callback)
-		throw new errors.ResinInvalidParameter('callback', callback, 'not a function')
-
-	exports.get name, (error, device) ->
-		return callback(error) if error?
-		return callback(null, !!device.is_online)
+	exports.get(name).then (device) ->
+		return !!device.is_online
```

Comment:
@Page- Cool, I'll remove the check. I also noticed that `is_online: undefined` in some cases (I think it happened before the device went online for the first time). Is that still the case?